src/librbd/image/RefreshRequest.cc

```diff
@@ -396,6 +415,71 @@ Context *RefreshRequest<I>::handle_v2_get_mutable_metadata(int *result) {
     m_incomplete_update = true;
   }
 
+  send_v2_get_parent();
+  return nullptr;
+}
+
+template <typename I>
+void RefreshRequest<I>::send_v2_get_parent() {
+  // NOTE: remove support when Mimic is EOLed
+  CephContext *cct = m_image_ctx.cct;
+  ldout(cct, 10) << this << " " << __func__ << ": legacy=" << m_legacy_parent
+                 << dendl;
+
+  librados::ObjectReadOperation op;
+  if (!m_legacy_parent) {
+    cls_client::parent_get_start(&op);
+    cls_client::parent_overlap_get_start(&op, CEPH_NOSNAP);
+  } else {
+    cls_client::get_parent_start(&op, CEPH_NOSNAP);
+  }
+
+  auto aio_comp = create_rados_callback<
+    RefreshRequest<I>, &RefreshRequest<I>::handle_v2_get_parent>(this);
+  m_out_bl.clear();
+  m_image_ctx.md_ctx.aio_operate(m_image_ctx.header_oid, aio_comp, &op,
+                                  &m_out_bl);
+  aio_comp->release();
+}
+
+template <typename I>
+Context *RefreshRequest<I>::handle_v2_get_parent(int *result) {
+  // NOTE: remove support when Mimic is EOLed
+  CephContext *cct = m_image_ctx.cct;
+  ldout(cct, 10) << this << " " << __func__ << ": r=" << *result << dendl;
+
+  auto it = m_out_bl.cbegin();
+  if (!m_legacy_parent) {
+    if (*result == 0) {
+      *result = cls_client::parent_get_finish(&it, &m_parent_md.spec);
+    }
+
+    if (*result == 0) {
+      *result = cls_client::parent_overlap_get_finish(&it,
+                                                      &m_parent_md.overlap);
+    }
+
+    if (*result == 0 && m_parent_md.overlap == 0) {
+      // TODO until parent ref is normalized, treat a zero-overlap as no
+      // parent linked to HEAD
+      m_parent_md.spec = {};
```

Comment:
@dillaman There is an issue with this. When resizing a clone to 0 and then removing, DetachChildRequest returns without detaching due to the empty parent. The removal succeeds but after this you can't unprotect the parent snapshot due to the stale reference. It is caught by python test_rbd.TestClone.test_resize_io:
```
zhuzha:~/ceph/ceph.trociny/build% CEPH_ARGS='--debug-rbd=30' RBD_FEATURES=127 PYTHONPATH=../src/test/pybind:../src/pybind:./lib/cython_modules/lib.2 LD_LIBRARY_PATH=./lib nosetests -v test_rbd:TestClone.test_resize_io     
 test_rbd.TestClone.test_resize_io ... ERROR

======================================================================
ERROR: test_rbd.TestClone.test_resize_io
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/nose/case.py", line 384, in tearDown
    try_run(self.inst, ('teardown', 'tearDown'))
  File "/usr/lib/python2.7/dist-packages/nose/util.py", line 471, in try_run
    return func()
  File "/ssd/ceph/ceph.trociny/src/test/pybind/test_rbd.py", line 1057, in tearDown
    self.image.unprotect_snap('snap1')
  File "rbd.pyx", line 2680, in rbd.Image.unprotect_snap (/home/mgolub/ceph/ceph.trociny/build/src/pybind/rbd/rbd.c:27401)
ImageBusy: [errno 16] error unprotecting snapshot image1@snap1
```